gadget/device_linux.go

```diff
@@ -54,11 +54,23 @@ func FindDeviceForStructure(ps *PositionedStructure) (string, error) {
 	var found string
 	var match string
 	for _, candidate := range candidates {
-		if !osutil.FileExists(candidate) {
```

Comment:
I'm not sure I get the aim 100% but I think this might be written in a slightly simpler way by using "osutil.IsSymlink()". I find the following (or some variation of it) a bit easier to read because its more linear and less nested. but maybe its more subtle and I miss something, in this case we need more tests :) With the change all tests still pass:
```go
                if !osutil.FileExists(candidate) {
			continue
		}
		if !osutil.IsSymlink(candidate) {
			return "", fmt.Errorf("candidate %v is not a symlink", candidate)
		}
		target, err := filepath.EvalSymlinks(candidate)
		if err != nil {
			return "", fmt.Errorf("cannot read device link: %v", err)
		}
		if found != "" && target != found {
                ...
```
(full diff: http://paste.ubuntu.com/p/Jd98SW3Xqk/)